src/Filters/Block.php

```diff
@@ -112,14 +112,8 @@ public function getFilterBlock(
             $idCategory = (int) Configuration::get('PS_HOME_CATEGORY');
         }
 
-        /* Get the filters for the current category */
-        $filters = $this->database->executeS(
-            'SELECT type, id_value, filter_show_limit, filter_type ' .
-            'FROM ' . _DB_PREFIX_ . 'layered_category ' .
-            'WHERE id_category = ' . $idCategory . ' ' .
-            'AND id_shop = ' . $idShop . ' ' .
-            'GROUP BY `type`, id_value ORDER BY position ASC'
-        );
+        // Get filters configured for the current query
+        $filters = $this->dataAccessor->getFiltersForQuery($this->query, $idShop);
```

Comment:
It's not really the responsibility of this class to get filters for queries, I think you could introduce a dedicated class that does that.